clang/lib/Sema/SemaChecking.cpp

```diff
@@ -4014,7 +4014,7 @@ void Sema::checkCall(NamedDecl *FDecl, const FunctionProtoType *Proto,
   if (FD)
     diagnoseArgDependentDiagnoseIfAttrs(FD, ThisArg, Args, Loc);
 
-  if (FD && FD->hasAttr<DeviceKernelAttr>())
+  if (FD && FD->hasAttr<DeviceKernelAttr>() && getLangOpts().isSYCL())
```

Comment:
Should that be checking for SYCLDevice, not SYCL? We don't need these checks for host, AFAIK.